### PR TITLE
ROC-5078: Dashboard unit tests

### PR DESCRIPTION
**Pull request by abshira**
### JIRA link

https://tools.hmcts.net/jira/browse/ROC-5078

### Change description
Claimant status : Claim creation 

Defendant: All full admission response payment choices & request for more time. 

### Work checklist

- [ ] Unit tests added where applicable
- [ ] Route tests added for new pages
- [ ] New pages included in a11y tests
- [ ] Task list and task completeness checks updated
- [ ] Check and send page updated
- [ ] Routes, page content and page flows of new features are toggled 
- [ ] UI changes look good on mobile
- [ ] Required Google Analytics events are being sent 

### Developer self-QA run statement

- [ x] I have clicked through the running application to see if all changes I made actually work.

### Does this PR introduce a breaking change?

- [ ] Yes
- [ x] No

### CODE REVIEW

File: src/test/data/entity/claimIssueData.ts [hunk 1]
```diff
@@ -0,0 +1,242 @@
...
+    freeMediation: 'no',
+    paymentIntention: {
+      paymentDate: '2019-02-11',
+      paymentOption: 'IMMEDIATELY'
+    },
+    features: ['admissions']
+  }
+}
+
+export const sampleInstallements = {
```
Comment (robertparkinson) [inline src/test/data/entity/claimIssueData.ts]: Should be sampleInstalments

File: src/test/features/dashboard/routes/index.ts [hunk 2]
```diff
@@ -13,6 +13,7 @@ import * as idamServiceMock from 'test/http-mocks/idam'
 import * as draftStoreServiceMock from 'test/http-mocks/draft-store'
 import * as claimStoreServiceMock from 'test/http-mocks/claim-store'
 import { checkAuthorizationGuards } from 'test/features/dashboard/routes/checks/authorization-check'
+import * as claimIssueData from '../../../data/entity/claimIssueData'
```
Comment (robertparkinson) [inline src/test/features/dashboard/routes/index.ts]: Don't use ../.. use test/

Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Update

File: src/test/data/entity/claimIssueData.ts [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+    freeMediation: 'no',
+    paymentIntention: {
+      paymentDate: '2019-02-11',
+      paymentOption: 'IMMEDIATELY'
+    },
+    features: ['admissions']
+  }
+}
+
+export const sampleInstalements = {
```
Comment (laurentaylor1) [inline src/test/data/entity/claimIssueData.ts]: Typo in `sampleInstalements` ->`sampleInstalments`

File: src/test/features/dashboard/routes/index.ts [hunk 4]
```diff
@@ -67,17 +61,154 @@ describe('Dashboard page', () => {
...
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'admitted part of the claim.'))
+        })
+        it('should render dashboard when defendant has respond with part admit & pay by a set date', async () => {
+          claimStoreServiceMock.resolveRetrieveBySampleDataDefendant(claimIssueData.partialAdmissionWithSetDate)
+
+          await request(app)
+            .get(Paths.dashboardPage.uri)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'admitted part of the claim.'))
+        })
```
Comment (KiranVarma02) [inline src/test/features/dashboard/routes/index.ts]: add line spacing..

File: src/test/features/dashboard/routes/index.ts [hunk 5]
```diff
@@ -67,17 +61,154 @@ describe('Dashboard page', () => {
...
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'You’ve admitted all of the claim and offered to pay the full amount in instalments.'))
+        })
+        it('should render dashboard when defendant has respond with part admit & pay immediately', async () => {
+          claimStoreServiceMock.resolveRetrieveBySampleDataDefendant(claimIssueData.partialAdmissionWithImmediatePaymentData)
+
+          await request(app)
+            .get(Paths.dashboardPage.uri)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'admitted part of the claim.'))
+        })
```
Comment (KiranVarma02) [inline src/test/features/dashboard/routes/index.ts]: line spacing...

File: src/test/features/dashboard/routes/index.ts [hunk 6]
```diff
@@ -67,17 +61,154 @@ describe('Dashboard page', () => {
...
+        it('should render dashboard when defendant has respond with a pay by a set date', async () => {
+          claimStoreServiceMock.resolveRetrieveBySampleDataDefendant(claimIssueData.sampleClaimIndividualVsIndividualFullAdmissionPayBySetDateObj)
+
+          await request(app)
+            .get(Paths.dashboardPage.uri)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'You’ve admitted all of the claim and offered to pay the full amount by'))
+
+        })
```
Comment (KiranVarma02) [inline src/test/features/dashboard/routes/index.ts]: line spacing

File: src/test/features/dashboard/routes/index.ts [hunk 7]
```diff
@@ -67,17 +61,154 @@ describe('Dashboard page', () => {
...
+        it('should render dashboard when defendant has respond with a pay immediately admission', async () => {
+          claimStoreServiceMock.resolveRetrieveBySampleDataDefendant(claimIssueData.sampleClaimIndividualVsIndividualFullAdmissionPayImmediatelyObj)
+
+          await request(app)
+            .get(Paths.dashboardPage.uri)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'You’ve admitted all of the claim and said you’ll pay the full amount immediately.'))
+
+        })
```
Comment (KiranVarma02) [inline src/test/features/dashboard/routes/index.ts]: line spacing...

File: src/test/features/dashboard/routes/index.ts [hunk 8]
```diff
@@ -67,17 +61,154 @@ describe('Dashboard page', () => {
...
+        it('should render dashboard when defendant has respond with a pay by repayment plan', async () => {
+          claimStoreServiceMock.resolveRetrieveBySampleDataDefendant(claimIssueData.sampleClaimIndividualVsIndividualFullAdmissionRepaymentPlanObj)
+
+          await request(app)
+            .get(Paths.dashboardPage.uri)
+            .set('Cookie', `${cookieName}=ABC`)
+            .expect(res => expect(res).to.be.successful.withText('000MC050', 'You’ve admitted all of the claim and offered to pay the full amount in instalments.'))
+        })
```
Comment (KiranVarma02) [inline src/test/features/dashboard/routes/index.ts]: line spacing..

File: src/test/features/dashboard/routes/index.ts (hunk 6 above)
Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Resolved

File: src/test/features/dashboard/routes/index.ts (hunk 8 above)
Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Resolved

File: src/test/features/dashboard/routes/index.ts (hunk 7 above)
Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Resolved

File: src/test/features/dashboard/routes/index.ts (hunk 5 above)
Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Resolved

File: src/test/features/dashboard/routes/index.ts (hunk 4 above)
Comment (abshira) [reply src/test/features/dashboard/routes/index.ts]: Resolved

File: src/test/data/entity/claimIssueData.ts [hunk 9]
```diff
@@ -0,0 +1,242 @@
...
+  response: {
+    responseType: 'PART_ADMISSION',
+    freeMediation: 'no',
+    defence: 'i have paid more than enough',
+    ...sampleInstalements,
+    ...sampleStatementOfMeans,
+    amount: 50
+  },
+  features: ['admissions']
+}
```
[the remark this answers is not in the document]
Comment (abshira) [reply src/test/data/entity/claimIssueData.ts]: resolved - added in responsedata.ts

File: src/test/data/entity/claimIssueData.ts [hunk 10]
```diff
@@ -0,0 +1,242 @@
...
+  submitterId: '1',
+  submitterEmail: 'claimant@example.com',
+  externalId: '400f4c57-9684-49c0-adb4-4cf46579d6da',
+  defendantId: '123',
+  referenceNumber: '000MC050',
+  createdAt: MomentFactory.currentDate(),
+  issuedOn: MomentFactory.currentDate(),
+  totalAmountTillToday: 200,
+  totalAmountTillDateOfIssue: 200,
+  moreTimeRequested: true,
```
[the remark this answers is not in the document]
Comment (abshira) [reply src/test/data/entity/claimIssueData.ts]: This refactoring will be part of another story via Zaki

File: src/test/data/entity/claimIssueData.ts [hunk 11]
```diff
@@ -0,0 +1,242 @@
...
+    } as Interest,
+    reason: 'Interest Reason',
+    feeAmountInPennies: 2500,
+    timeline: { rows: [{ date: 'a', description: 'b' }] }
+  },
+  responseDeadline: MomentFactory.currentDate().add(19, 'days'),
+  features: ['admissions']
+}
+
+export const sampleStatementOfMeans = {
```
[the remark this answers is not in the document]
Comment (abshira) [reply src/test/data/entity/claimIssueData.ts]: Resolved - removed some json obj to the correct pages

File: src/test/data/entity/claimIssueData.ts [hunk 12]
```diff
@@ -0,0 +1,242 @@
...
+      name: 'John Doe',
+      address: {
+        line1: 'line1',
+        line2: 'line2',
+        city: 'city',
+        postcode: 'bb127nq'
+      }
+    }
+  ]
+}
```
[the remark this answers is not in the document]
Comment (abshira) [reply src/test/data/entity/claimIssueData.ts]: resolved

**Pull request merged**